src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -4446,10 +4446,10 @@ private Object stringSubstringMultiByte(DynamicObject string, int beg, int chara
 
             int p;
             int s = 0;
```

Comment:
Minor and existing before: `s` is never assigned so could be replaced by 0 at all usages.